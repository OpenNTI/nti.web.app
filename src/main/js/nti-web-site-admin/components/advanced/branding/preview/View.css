.preview-dialog {
	width: 768px;
}

.preview-dialog ul {
	list-style: none;
	padding: 0;
	margin: 0;
}

.preview-dialog > ul > li {
	box-shadow: 0 2px 5px 0 rgba(0, 0, 0, 0.15);
	margin-bottom: 2rem;
}

.title {
	color: white;
	font-size: 1.125rem;
	margin: 0.5em 0;
}

.preview-dialog :global(.icon-light-x) {
	background: white;
	border-radius: 50%;
	font-size: 1.75rem;
	padding: 0.75rem;
	position: absolute;
	right: calc(100% + 2rem);
	cursor: pointer;
}

.controls {
	text-align: center;
}
