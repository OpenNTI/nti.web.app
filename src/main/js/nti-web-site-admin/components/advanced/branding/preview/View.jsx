import React from 'react';
import PropTypes from 'prop-types';
import {Button, Prompt} from '@nti/web-commons';
import classnames from 'classnames/bind';

import t from '../sections/assets/strings';
import {types} from '../sections/assets/constants';

import getWidget from './widgets';
import styles from './View.css';

const cx = classnames.bind(styles);

export default function Preview ({onSave, onClose}) {
	return (
		<Prompt.Dialog>
			<div className={cx('preview-dialog')}>
				<span className="icon-light-x" onClick={onClose} />
				<ul>
					{
						Object.values(types).map(type => {
							const Widget = getWidget(type);
							return !Widget ? null : (
								<li key={type}>
									<div className={cx('title')}>{t(['types', type, 'title'])}</div>
									<Widget />
								</li>
							);
						})
					}
				</ul>
				{ onSave && (
					<div className={cx('controls')}>
						<Button onClick={onSave}>Apply Changes</Button>
					</div>
				)}
			</div>
		</Prompt.Dialog>
	);
}
