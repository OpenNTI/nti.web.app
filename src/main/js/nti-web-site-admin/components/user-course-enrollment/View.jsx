import {Router, Route} from '@nti/web-routing';// eslint-disable-line

import Overview from './overview';
import Reports from './reports';
import Progress from './progress';
import Frame from './Frame';

export default Router.for([
	Route({path: '/reports', component: Reports, name: 'site-admin.users.user-course-enrollment.reports'}),
	Route({path: '/progress', component: Progress, name: 'site-admin.users.user-course-enrollment.progress'}),
	Route({path: '/', component: Overview, name: 'site-admin.users.user-course-enrollment.overview'})
], {frame: Frame});
