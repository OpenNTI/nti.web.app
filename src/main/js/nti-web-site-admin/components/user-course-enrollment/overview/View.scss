@import '~@nti/style-common';

.site-admin-user-enrollment-overview {
	.date-info {
		display: flex;

		.joined {
			margin: 0 10px 10px 0;
			width: 370px;

			.site-admin-card {
				background-color: #3fb34f;
				color: white;
			}
		}

		.last-activity {
			width: 370px;
		}
	}

	.active-times-widget {
		margin: 10px 0 0;
	}

	.activity-and-progress {
		display: flex;
		flex-direction: row;
		align-items: flex-start;

		.user-course-enrollment-progress {
			margin: 10px 0 10px 10px;
			width: 240px;

			.circular-progress {
				width: 200px;
				margin: 0 auto;
			}
		}
	}
}
