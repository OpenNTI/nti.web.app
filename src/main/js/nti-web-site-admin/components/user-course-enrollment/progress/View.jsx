import React from 'react';
import PropTypes from 'prop-types';
import {Progress} from '@nti/web-course';

import Card from '../../common/Card';

SiteAdminUserEnrollmentProgress.propTypes = {
	course: PropTypes.object,
	enrollment: PropTypes.object
};
export default function SiteAdminUserEnrollmentProgress ({course, enrollment}) {
	return (
		<Card>
			{course && enrollment && (
				<Progress.Overview course={course} enrollment={enrollment} singleItem />
			)}
		</Card>
	);
}
