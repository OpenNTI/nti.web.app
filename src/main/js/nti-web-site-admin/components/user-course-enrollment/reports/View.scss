@import '~@nti/style-common';

.site-admin-user-overview {
	@include clearfix;

	& > * {
		float: left;
		padding: 0 0.5rem 0.5rem 0;
	}

	.joined {
		width: 50%;

		.site-admin-date-value {
			background-color: var(--primary-green);
			color: white;
		}
	}

	.last-login {
		width: 50%;
	}
}
