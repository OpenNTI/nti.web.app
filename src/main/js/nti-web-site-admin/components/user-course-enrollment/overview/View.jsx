import './View.scss';
import React from 'react';
import PropTypes from 'prop-types';
import {Loading} from '@nti/web-commons';
import { Widgets } from '@nti/web-reports';

import DateValue from '../../common/DateValue';

import ActiveTimes from './ActiveTimes';
import LastActivity from './LastActivity';
import Progress from './Progress';

const { ActiveDays } = Widgets;

export default class SiteAdminUserEnrollmentView extends React.Component {
	static propTypes = {
		enrollment: PropTypes.object
	}

	render () {
		const { enrollment } = this.props;

		if(!enrollment) {
			return <Loading.Mask/>;
		}

		return (
			<div className="site-admin-user-enrollment-overview">
				<div className="date-info">
					<div className="joined">
						<DateValue date={enrollment.getCreatedTime()} label="Enrolled" />
					</div>
					<div className="last-activity">
						<LastActivity enrollment={enrollment}/>
					</div>
				</div>
				<ActiveDays entity={enrollment}/>
				<div className="activity-and-progress">
					<ActiveTimes enrollment={enrollment}/>
					{enrollment.CourseProgress && (<Progress enrollment={enrollment} />)}
				</div>
			</div>
		);

	}

}
