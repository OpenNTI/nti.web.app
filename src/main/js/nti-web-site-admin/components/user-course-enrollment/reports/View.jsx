import './View.scss';
import React from 'react';
import PropTypes from 'prop-types';

import { List } from '@nti/web-reports';

import Card from '../../common/Card';

SiteAdminUserEnrollmentReports.propTypes = {
	enrollment: PropTypes.object,
};
export default function SiteAdminUserEnrollmentReports({ enrollment }) {
	return enrollment ? (
		<Card>
			<List context={enrollment} />
		</Card>
	) : null;
}
