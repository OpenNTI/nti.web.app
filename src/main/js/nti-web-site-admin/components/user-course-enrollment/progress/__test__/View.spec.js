import renderer, { act } from 'react-test-renderer';

import View from '../View';

/* eslint-env jest */
describe('Site admin user course enrollment progress view', () => {
	test('Basic render test', async () => {
		const course = {
			hasLink: () => true,
			getContentDataSource: () => ({
				loadPage: () => {},
				getTotalCount: () => 0,
			}),
		};

		const enrollment = {};

		// this component will show an "unable to load" message in the snapshot, which is ok.  The underlying Progress
		// widget (in nti-web-course) should be tested on its own
		let cmp;
		act(() => {
			cmp = renderer.create(
				<View course={course} enrollment={enrollment} />
			);
		});

		const tree = cmp.toJSON();

		expect(tree).toMatchSnapshot();

		cmp.unmount();
	});

	test('Basic render test (no course)', async () => {
		const enrollment = {};

		const cmp = renderer.create(<View enrollment={enrollment} />);

		const tree = cmp.toJSON();

		expect(tree).toMatchSnapshot();
	});

	test('Basic render test (no enrollment)', async () => {
		const course = {};

		let cmp;
		act(() => {
			cmp = renderer.create(<View course={course} />);
		});

		const tree = cmp.toJSON();

		expect(tree).toMatchSnapshot();
		cmp.unmount();
	});
});
