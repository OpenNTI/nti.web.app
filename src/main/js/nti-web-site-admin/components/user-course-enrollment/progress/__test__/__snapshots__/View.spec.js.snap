// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Site admin user course enrollment progress view Basic render test (no course) 1`] = `
<div
  className="site-admin-card"
/>
`;

exports[`Site admin user course enrollment progress view Basic render test (no enrollment) 1`] = `
<div
  className="site-admin-card"
/>
`;

exports[`Site admin user course enrollment progress view Basic render test 1`] = `
<div
  className="site-admin-card"
>
  <div
    className="progress-overview"
  >
    <div
      className="progress-overview-stats"
    >
      <div
        className="progress-overview-stats-charts large"
      >
        <div
          className="switcher"
        >
          <span
            className="chart"
          >
            Course Standing
          </span>
        </div>
        <div
          className="active-chart"
        >
          <div
            className="progress-overview-charts-course-standing"
          >
            <div
              className="error-container"
            >
              <div
                className="error"
              >
                Unable to load progress statistics.
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
